src/metabase/lib/fe_util.cljc

```diff
@@ -108,3 +111,53 @@
 
       _
       (lib.metadata.calculation/display-name query stage-number filter-clause))))
+
+(defn- query-dependents
+  [metadata-providerable query-or-join]
+  (let [base-stage (first (:stages query-or-join))
+        database-id (:database query-or-join -1)]
+    (concat
+     (when (pos? database-id)
+       [{:type :database, :id database-id}
+        {:type :schema,   :id database-id}])
```

Comment:
I always return `:database` and `:schema` in pairs, which seems a bit fishy. Either there should be situations when one of them is missing, or just one of them should be enough.

Reply:
I would rather drop `:schema` in the FE if it means the same as `:database` or figure out when these can be different. Going forward with this, feels like it will always be a source of confusion.